### ISSUE TITLE
Closing a window also exits the entire overview

### ISSUE THREAD

**Issue by mapleroyal** [p91paul/middleclickclose#40]
I assume this isn't intentional, but I don't see an option to toggle the behavior in the extension's settings.

Happens with both touchpad and mouse. Overview opened by pressing the meta key.

This is on a fresh install of vanilla arch, in a wayland session.

**Comment by dreua** [comment]
I've never seen this, can you try in an X Session?

**Comment by p91paul** [comment]
looks like https://github.com/p91paul/middleclickclose/issues/20, how did you install the extension? that bugged version shouldn't be available anywhere

**Comment by mapleroyal** [comment]
@p91paul 
> looks like #20, how did you install the extension? that bugged version shouldn't be available anywhere

Just via the official extensions website.

@dreua not at computer currently, and it doesn't boot right now anyways.

**Comment by mapleroyal** [comment]
@p91paul Still happening, this time on another brand new install of arch on a clean SSD, same hardware (XPS 9720).

**Comment by p91paul** [comment]
I can reproduce this only if I try to close some specific windows, like dirty text editors, that require confirmation to close, and I believe that is by design. I just released a new version on extension.gnome.org that does something slightly different and fixes #18, could you check if maybe this issue is fixed as well? I expect the update to come through tomorrow or in a few days max.

**Comment by mapleroyal** [comment]
> I can reproduce this only if I try to close some specific windows, like dirty text editors, that require confirmation to close, and I believe that is by design. I just released a new version on extension.gnome.org that does something slightly different and fixes #18, could you check if maybe this issue is fixed as well? I expect the update to come through tomorrow or in a few days max.

I certainly will. In the meantime, I should note that this has never NOT been an issue on any setup I do, including two more machines in the last week, one a laptop and one a desktop. Here's to hoping the update changes something!

**Comment by mapleroyal** [comment]
> I can reproduce this only if I try to close some specific windows, like dirty text editors, that require confirmation to close, and I believe that is by design. I just released a new version on extension.gnome.org that does something slightly different and fixes #18, could you check if maybe this issue is fixed as well? I expect the update to come through tomorrow or in a few days max.

I found the cause, but not the specific primary cause. The bug occurs when Dash To Panel is active. Next I'll reset the dash to panel settings and see when the problem starts.

What's the new version number? The gnome extensions website and this repo both don't show any version release date history so I'm not sure if I'm on the new version to test it.

**Comment by p91paul** [comment]
latest version is 27, you can check the metadata.json file in $HOME/.local/share/gnome-shell/extensions/middleclickclose@paolo.tranquilli.gmail.com

**Comment by mapleroyal** [comment]
Any idea when this will be pushed to the gnome extensions website? I'm on version 56, but the website does does nothing to indicate when that version was released. This is still an issue in 56.

**Comment by mapleroyal** [comment]
> latest version is 27, you can check the metadata.json file in $HOME/.local/share/gnome-shell/extensions/[middleclickclose@paolo.tranquilli.gmail.com](mailto:middleclickclose@paolo.tranquilli.gmail.com)

Any idea when this will be pushed to the gnome extensions website? I'm on version 56, but the website does does nothing to indicate when that version was released. This is still an issue in 56.

**Comment by p91paul** [comment]
sorry but there is no version 56 of this extension. In the latest post you said this happens when using dash to panel. I do not use that extension, did you try reporting the bug to them? looks like they are also messing with the overview behavior like this extension does.

**Comment by mapleroyal** [comment]
Apologies, that comment was meant for the issue in the dash to panel repo. I'm sorry.

**Comment by gtsiam** [comment]
Judging by the [issue](https://github.com/home-sweet-gnome/dash-to-panel/issues/1876) you made on dash to panel, I'm guessing this is fixed for dash to panel, but I'll give it a test once gnome 45 support lands for:
- [ ] Dash to dock (https://github.com/micheleg/dash-to-dock/pull/2070)
- [ ] Dash to panel (https://github.com/home-sweet-gnome/dash-to-panel/pull/1939)

**Comment by gtsiam** [comment]
This seems to be fixed. If you encounter the issue again, please open another issue.
